core/typedefs.h

```diff
@@ -69,6 +69,47 @@
 
 #endif
 
+// No discard allows the compiler to flag warnings if we don't use the return value of functions / classes
+#ifndef _NO_DISCARD_
+// c++ 17 onwards
+#if __cplusplus >= 201703L
+#define _NO_DISCARD_ [[nodiscard]]
+#else
```

Comment:
Are these C++17 cases necessary? C++14 is explicitly selected on `3.x`.

Reply:
No, and @akien-mga did get me to remove the pre - c++ 17 defines in the Godot 4 version, so it is an option to remove this part.

However personally I would be tempted to err on the side of leaving this for forward compatibility - should we decide in future to compile for c++ 17, it should just work (TM) even better in the same way as 4.x, and no need to revisit the code (and it is fresh in mind at the moment, but may not be in e.g. a years time).

Reply:
Yeah it's good to have it included if we ever decide to move to C++17 in `3.x`, then it will just work (and we can eventually cleanup the unneeded non-standard attributes).